tests/tests_common.py

```diff
@@ -699,6 +699,87 @@ def test_base_data_type(self):
         self.assertEqual(ua_utils.get_base_data_type(d), self.opc.get_node(ua.ObjectIds.Structure))
         self.assertEqual(ua_utils.data_type_to_variant_type(d), ua.VariantType.ExtensionObject)
 
-
-
-
+    def test_ua_method_unique(self):
+        def unique_response(parent):
+            return 42
+
+        decorated_unique_response = uamethod(unique_response)
+        response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+
+        self.assertEqual(len(response), 1)
+        self.assertEqual(type(response[0]), ua.Variant)
+
+    def test_ua_method_multiple(self):
+        def unique_response(parent):
+            return 42, 49
+
+        decorated_unique_response = uamethod(unique_response)
+        response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+
+        self.assertEqual(len(response), 2)
+        self.assertEqual(type(response[0]), ua.Variant)
+
+    def test_ua_method_complex(self):
+        register_extension_object(KeyValuePair)
+
+        def unique_response(parent):
+            return KeyValuePair("Key", "Value")
+
+        decorated_unique_response = uamethod(unique_response)
+        response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+
+        self.assertEqual(len(response), 1)
+        self.assertEqual(type(response[0]), ua.Variant)
+
+    def test_ua_method_complex_multiple(self):
+        register_extension_object(KeyValuePair)
+
+        def unique_response(parent):
+            return [KeyValuePair("Key", "Value"),
+                    KeyValuePair("Hello", "World")]
+
+        decorated_unique_response = uamethod(unique_response)
+        response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+
+        self.assertEqual(len(response), 2)
+        self.assertEqual(type(response[0]), ua.Variant)
+
+    # TODO: Fix the test
+    # def test_decode_custom_object(self):
+    #     register_extension_object(KeyValuePair)
+    #
+    #     def unique_response(parent):
+    #         return KeyValuePair("Key", "Value")
+    #
+    #     decorated_unique_response = uamethod(unique_response)
+    #     response = decorated_unique_response(ua.NodeId("ServerMethod", 2))
+    #
+    #     self.assertEqual(len(response), 1)
+    #     self.assertEqual(type(response[0]), ua.Variant)
+    #
+    #     key_value = uabin.unpack_uatype_array(response[0].VariantType, io.BytesIO(response[0].to_binary()))
+    #     self.assertEqual(key_value.key, "Key")
+    #     self.assertEqual(key_value.value, "Value")
+
+
+class KeyValuePair(object):
```

Comment:
would be great to give that class a name like MyCustomStructure. It took me a while to understand that this was just an example